polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistOps.td

```diff
@@ -113,6 +113,19 @@ def Pointer2MemrefOp : Polygeist_Op<"pointer2memref", [
   }];
 }
 
+def AddrSpaceCastOp : Polygeist_Op<"addrspacecast", [
+  ViewLikeOpInterface, Pure
+]> {
+  let summary = "Cast address space of a memref";
+
+  let arguments = (ins AnyMemRef : $source);
+  let results = (outs AnyMemRef : $result);
+
+  let extraClassDeclaration = [{
+    ::mlir::Value getViewSource() { return getSource(); }
+  }];
```

Comment:
Maybe add verifier to check the only memory space changes between input and output `MemRef` types? We should assert layout, pointed type and shape remain unchanged.

Reply:
I see MLIR has `CastOpInterface`. Seems to me that we should add that interface to this operation as well.